src/components/Map/Map.tsx

```diff
@@ -40,13 +40,16 @@ export default function Map(props: MapProps): JSX.Element {
   const [geoData, setGeoData] = useState();
   const [layerIds, setLayerIds] = useState<string[]>([]);
   const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
+  const [zoomBegin, setZoomBegin] = useState<boolean>(false);
+  const [zoomEnd, setZoomEnd] = useState<boolean>(false);
   const mapRef = useRef<any | null>(null);
 
   const onMapLoad = useCallback(() => {
     const { bbox } = options;
     // fit to bounding box
     if (mapRef?.current !== undefined) {
       const mapInstance: any = mapRef.current.getMap();
+      setZoomBegin(true);
```

Comment:
should we also `setZoomEnd(false)` here?

Reply:
Oh I do see now that this should only run on map load and that zoomEnd is initialized to false 🤦 Apologies. I guess it was not clear to me when zoomEnd is ever set to false (other than when initialized). Is there not an "onZoomBegin" handler to mirror the "onZoomEnd" one?

Reply:
Your comment was totally valid, explicitly setting zoomEnd made the code readable and also made the assumptions very clear.